plugin/trino-iceberg/pom.xml

```diff
@@ -16,6 +16,15 @@
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
         <dep.iceberg.version>0.12.0</dep.iceberg.version>
+        <!--
+         Project's default for air.test.parallel is 'methods'. By design, 'classes' makes TestNG run tests from one class in a single thread.
+         As a side effect, it prevents TestNG from initializing multiple test instances upfront, which happens with 'methods'.
+         A potential downside can be long tail single-threaded execution of a single long test class.
+         This change is due to MVN Surefire issue which is addressed here https://github.com/apache/maven-surefire/pull/407
+         -->
+        <air.test.parallel>classes</air.test.parallel>
+        <!-- If two MinIO based test will run in parallel OOM will occur -->
+        <air.test.thread-count>1</air.test.thread-count>
```

Comment:
is this actually effective?

i experimented with that in https://github.com/findepi/surefire-testng-ordering-example/blob/5ca285f80811bcbddfb8e2d5c9c203d6a6427f45/src/test/java/localhost/surefiretestngordering/Main.java#L30-L31
and it seems that switching to "classes" and thread count 1 isn't a workaround

```
[thread 21] TestClass1 / Base.setupAllocateResources: resources allocated: 1
[thread 21] TestClass2 / Base.setupAllocateResources: resources allocated: 2
[thread 21] TestClass3 / Base.setupAllocateResources: resources allocated: 3
[thread 21] TestClass4 / Base.setupAllocateResources: resources allocated: 4
[thread 21] TestClass1 / Base.tearDownReleaseResources: resources allocated: 3
[thread 21] TestClass2 / Base.tearDownReleaseResources: resources allocated: 2
[thread 21] TestClass3 / Base.tearDownReleaseResources: resources allocated: 1
[thread 21] TestClass4 / Base.tearDownReleaseResources: resources allocated: 0
```

Reply:
It worked in my case. But this was only to prevent 2 test running at same time.
As it's test it self got large memory consumption. So this would just prevent them to run in parallel.
Now it will not be required as we reduced memory usage.

Reply:
Preventing two tests from running at the same time surely does reduce memory needs and this is what `air.test.thread-count` is about.

However, i would expect more significant memory savings from preventing multiple `QueryRunner`s from being live at the same time. This is what `air.test.parallel` `classes` is supposed to be doing, but was not doing that in my tests on a side.
@findinpath i know we used that for Delta Lake tests with success, but per above it seems not effective. What am i missing?

Reply:
I think we are talking on the same thing but from different angles :).
I'm not saying it's not working. It reduced memory needs, but still single test running consumed 2,7GB of RAM which made 2 of them running in parallel exceed 3GB of allocated memory.
Once I've used suggested `hive.s3.streaming.part-size=5MB` memory needs for single test run dropped to around 1GB which allows to run them in parallel.
In this particular case QueryRunner memory allocation was not an issue. As it was just test scenario making memory usage large which was actually pretty fast cleaned up during the test. But if they happen to run together there was no chance to make that GC.